Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +10,236 @@ public static class BalanceStats
+
+                    if (isBeforeBalanceValid && isAfterBalanceValid)
+                    {
+                        var loss = endAmount - beginAmount;
+                        if (loss < biggestLoss)
+                        {
+                            biggestLoss = loss;
+                        }
+                    }
+                }
+            }
+
+            return biggestLoss;
         }
 
         /// <summary>
         /// Return name and current money of the richest person.
         /// </summary>
         public static string FindRichestPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (IsArrayNullOrEmpty(peopleAndBalances))
+            {
+                return "N/A.";
+            }
+
+            var maxBalance = 0m;
+            var name = "";
+
+            for (var i = 0; i < peopleAndBalances.Length; i++)
+            {
+                var peopleBalances = peopleAndBalances[i].Split(",");
+                var isBalanceValid = decimal.TryParse(peopleBalances[peopleBalances.Length - 1], out var balance);
+                
+                if (isBalanceValid && balance >= maxBalance)
+                {
+                    if (balance > maxBalance)
+                    {
+                        name = "";
+                    }
+                    if (name != "")
+                    {
+                        name += ", ";
+                    }
+                    name += peopleBalances[0];
+
+                    maxBalance = balance;
+                }
+            }
+
+            SetMoneyFormat();
+            var maxBalanceNoComma = RemoveComma($"{maxBalance:C0}");
+            var word = name.Contains(", ") ? "are" : "is";
+            var word2 = name.Contains(", ") ? "people" : "person";
+            return $"{ReplaceLastComma(name)} {word} the richest {word2}. {maxBalanceNoComma}.";
         }
 
         /// <summary>
         /// Return name and current money of the most poor person.
         /// </summary>
         public static string FindMostPoorPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (IsArrayNullOrEmpty(peopleAndBalances))
+            {
+                return "N/A.";
+            }
+
+            var minBalance = decimal.MaxValue;
+            var name = "";
+
+            for (var i = 0; i < peopleAndBalances.Length; i++)
+            {
+                var peopleBalances = peopleAndBalances[i].Split(",");
+                var isBalanceValid = decimal.TryParse(peopleBalances[peopleBalances.Length - 1], out var balance);
+
+                if (isBalanceValid && balance <= minBalance)
+                {
+                    if (balance < minBalance)
+                    {
+                        name = "";
+                    }
+                    if (name != "")
+                    {
+                        name += ", ";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use a StringBuilder instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4955013)